### PR TITLE
Unfollow authors via the #179

### LINKED ISSUE #179: Handle Remote Author Follow, Unfollow and Friend
Show Notification in inbox. Show status in author card: following/ friend.
Need to use two servers to test func.
Check the case where a remote server is not responding.

### PR DESCRIPTION
**Pull request by debecerra**
### Summary
- Resolves #179 
- User can follow, follow back (friend) and unfollow (this one is new) remote users. 
- Can unfollow using API
- Create users on different servers, become friends then send a friend's post and the other will see it. 

### Outstanding issues:
- /author page for remote author is showing API instead of a nice UI. This is a bug. This means you can't see the status of a follower on the author profile card like we can for local authors. However, you can check their follow/friend status by creating a post and seeing if the remote server receives it.
- working on fixing this so we can actually unfollow via UI

A few sample screenshots because this is a pain to set up and test. Note the hosts. Note that local is on the inbox page and dummy is their home page viewing the post they just created.

host: 127.0.0.1:8000
username: local
password: supersecret

diego receives a post from leo

image

host: 127.0.0.1:8080
username: dummy
password: supersecret

leo sent diego a friend post

image

### CODE REVIEW

**Comment by debecerra** [comment]
Going to open this PR as-is. Local unfollows work, and the API for unfollowing works so other groups can use it. The only thing that doesn't work is our UI for remote unfollows, we can address this in a new PR.

File: code/socialDistribution/requests.py [hunk 1]
```diff
@@ -117,11 +117,20 @@ def delete(url, params=None):
...
     headers = {
-        "Accept": "application/json",
-        "REFERER": HOST
+        "Accept": "application/json"
     }
 
-    response = requests.delete(url, headers=headers, params=params)
+    # ref: https://stackoverflow.com/questions/15431044/can-i-set-max-retries-for-requests-request - datashaman
+    # 'Can I set max_retries for requests.request?'
+    session = requests.Session()
```
Comment (JawadHossain) [inline code/socialDistribution/requests.py:125]: consider wrapping this in try-except as well as these lines could fail for many reasons, similar to get above. thanks

**Review by JawadHossain — APPROVED** [review]
Looks great, thanks, Diego!
On clicking remote authors I originally set it to take to remote author API url instead of author posts profile like our local authors since we can't be sure what frontend profile remote groups use. I think we can find a better fix for this.
but this shouldn't affect the follow, unfollow, friend tags on an author card right?

**Review by olivier-vadiaval — APPROVED** [review]
Thanks a lot Diego! 💯

**Comment by debecerra** [comment]
> Looks great, thanks, Diego! On clicking remote authors I originally set it to take to remote author API url instead of author posts profile like our local authors since we can't be sure what frontend profile remote groups use. I think we can find a better fix for this. but this shouldn't affect the follow, unfollow, friend tags on an author card right?

I think the intended functionality is that a local user will use our frontend to view the data for the remote profile (display name, click to follow, profile picture, etc). So the missing component is that we need to GET from the API and parse that data to display our UI, instead of just redirecting straight to the API.

**Pull request merged**